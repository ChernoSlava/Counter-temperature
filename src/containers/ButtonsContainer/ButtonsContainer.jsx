import React, { useContext } from "react";
import { Button } from "@components"
import { CounterContext } from "../../contexts";
import { actionType } from "@constants";

export const ButtonsContainer = () => {
    const {dispatch} = useContext(CounterContext);

    return (
        <>
            <Button 
                title="+" 
                type='plus' 
                onClick={() => dispatch({ type: actionType.plus })}/>
            <Button 
                title="-" 
                type='minus' 
                onClick={() => dispatch({ type: actionType.minus })}/>
        </>
    )
}
