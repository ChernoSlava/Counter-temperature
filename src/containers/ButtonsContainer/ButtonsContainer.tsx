import React from "react";
import { useDispatch } from "react-redux";

import { Button } from "@components"
import { buttonsTitle } from "@constants";
import { MinusAction, PlusAction } from '@store';

export const ButtonsContainer: React.FC = (): JSX.Element => {
  const dispatch = useDispatch();

  return (
    <>
      <Button 
        title={buttonsTitle.titleMinus} 
        type='minus'
        onClick={() => dispatch(MinusAction())}
      />
      <Button 
        title={buttonsTitle.titlePlus}
        type='plus'
        onClick={() => dispatch(PlusAction())}
      />
    </>
  )
}
