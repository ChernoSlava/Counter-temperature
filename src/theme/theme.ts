export const theme = {
  colors: {
    grey10: '#696867',

    blue10: 'rgb(19, 28, 192)',
    blue20: 'rgb(72, 133, 218)',
    blue30: 'rgb(98, 138, 196)',
    blue40: 'rgb(0, 69, 166)',

    red10: 'rgb(136, 29, 29)',
    red20: 'rgb(252, 38, 0)',
    red30: 'rgb(215, 104, 84)',

    black: 'black',
    white: 'white',
  },
};

export type ThemeType = typeof theme;
