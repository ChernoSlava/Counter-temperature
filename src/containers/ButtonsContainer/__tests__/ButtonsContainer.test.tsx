import { ButtonsContainer } from "../ButtonsContainer";
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Store, AnyAction } from '@reduxjs/toolkit';

import { theme } from '../../../theme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { InitialCounter } from "@store";

const mockStore = configureStore();

const getUI = (store: Store<any, AnyAction>): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <ButtonsContainer />
      </Provider>
    </ThemeProvider>
  );
};

describe('ButtonsContainer test', () => {
  it('Should render', () => {
    const { container } = render(getUI(mockStore({
      ...InitialCounter,
    }),
    ))
    expect(container).toMatchSnapshot();
  })

  it('Should call Plus action', () => {
    const store = mockStore({
      ...InitialCounter,
    });

    const {getByText} = render(getUI(store));

    const button = getByText('+');

    fireEvent.click(button);

    expect(store.getActions()).toMatchSnapshot();
  })

  it('Should call Minus action', () => {
    const store = mockStore({
      ...InitialCounter,
    });

    const {getByText} = render(getUI(store));

    const button = getByText('-');

    fireEvent.click(button);

    expect(store.getActions()).toMatchSnapshot();
  })
})