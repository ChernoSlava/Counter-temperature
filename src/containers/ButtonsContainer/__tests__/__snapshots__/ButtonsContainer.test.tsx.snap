// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsContainer test Should call Minus action 1`] = `
[
  {
    "payload": undefined,
    "type": "Counter_Temperature/minus",
  },
]
`;

exports[`ButtonsContainer test Should call Plus action 1`] = `
[
  {
    "payload": undefined,
    "type": "Counter_Temperature/plus",
  },
]
`;

exports[`ButtonsContainer test Should render 1`] = `
.c0 {
  width: 70px;
  height: 70px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 40px;
  color: white;
  border: 1px solid white;
  background-color: #696867;
}

.c0:hover {
  background-color: rgb(19,28,192);
  -webkit-transition: .7s;
  transition: .7s;
}

.c1 {
  width: 70px;
  height: 70px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 40px;
  color: white;
  border: 1px solid white;
  background-color: #696867;
}

.c1:hover {
  background-color: rgb(136,29,29);
  -webkit-transition: .7s;
  transition: .7s;
}

<div>
  <button
    class="c0"
  >
    -
  </button>
  <button
    class="c1"
  >
    +
  </button>
</div>
`;
